### PR TITLE
Fix cppcheck 1.87 warnings in MDAlgorithms

### PR DESCRIPTION
**Pull request by soininen**
This PR fixes most `cppcheck` 1.87 issues in the MDAlgorithms module. Some cases where `cppcheck` suggests using an STL algorithm instead of a raw loop were left as-is since these loops were really simple and perhaps not worth the trouble of rewriting.

**To test:**

Code review

A partial fix to #22912.

*This does not require release notes* because this is an internal change.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/MDAlgorithms/src/BoxControllerSettingsAlgorithm.cpp [hunk 1]
```diff
@@ -37,8 +37,9 @@ void BoxControllerSettingsAlgorithm::initBoxControllerProps(
                    tokenizer::TOK_IGNORE_EMPTY | tokenizer::TOK_TRIM);
   std::vector<int> valueVec;
   valueVec.reserve(values.count());
-  for (const auto &value : values)
-    valueVec.push_back(boost::lexical_cast<int>(value));
+  std::transform(
```
Comment (martyngigg) [inline Framework/MDAlgorithms/src/BoxControllerSettingsAlgorithm.cpp]: Nice.

**Review by martyngigg — DISMISSED** [review]
Good set of changes.

File: Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp [hunk 2]
```diff
@@ -483,14 +483,13 @@ void IntegrateEllipsoids::exec() {
...
       sateprincipalaxis3;
-  V3D peak_q;
   for (size_t i = 0; i < n_peaks; i++) {
     V3D hkl(peaks[i].getIntHKL());
     V3D mnp(peaks[i].getIntMNP());
 
     if (Geometry::IndexingUtils::ValidIndex(hkl, 1.0) ||
         Geometry::IndexingUtils::ValidIndex(mnp, 1.0)) {
-      peak_q = peaks[i].getQLabFrame();
+      V3D peak_q = peaks[i].getQLabFrame();
```
Comment (DanNixon) [inline Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp]:
```suggestion
      const V3D peak_q = peaks[i].getQLabFrame();
```

File: Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp [hunk 3]
```diff
@@ -404,8 +404,8 @@ void IntegratePeaksMD2::integrate(typename MDEventWorkspace<MDE, nd>::sptr ws) {
             useOnePercentBackgroundCorrection);
 
         // Relative volume of peak vs the BackgroundOuterRadius sphere
-        double ratio = (PeakRadius / BackgroundOuterRadius);
-        double peakVolume = ratio * ratio * ratio;
+        double radiusRatio = (PeakRadius / BackgroundOuterRadius);
+        double peakVolume = radiusRatio * radiusRatio * radiusRatio;
```
Comment (DanNixon) [inline Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp]: Can these be `const`?

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp (hunk 3 above)
Comment (soininen) [reply Framework/MDAlgorithms/src/IntegratePeaksMD2.cpp]: Yes.

File: Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp (hunk 2 above)
Comment (soininen) [reply Framework/MDAlgorithms/src/IntegrateEllipsoids.cpp]: I `const`ed this and did some general refactoring in the loop.

**Review by DanNixon — DISMISSED** [review]

**Review by DanNixon — APPROVED** [review]

**Pull request merged**